### ISSUE TITLE
Documentation error for sudo sh

### ISSUE THREAD

**Issue by candlerb** [mitchellh/packer#3985]
At https://www.packer.io/docs/provisioners/shell.html it gives the following example of `execute_command` for using sudo:

```
"echo 'packer' | {{ .Vars }} sudo -E -S sh '{{ .Path }}'"
```

However that's not right. It should be either:

```
"echo 'packer' | {{ .Vars }} sudo -E -S '{{ .Path }}'"
```

or:

```
"echo 'packer' | {{ .Vars }} sudo -E -S sh -e '{{ .Path }}'"
```

With the setting shown, any errors in the script are silently ignored, because invoking `sh` + script ignores the shebang line, and no `-e` is being passed here.

Since packer sets the executable flag, the example is best fixed by removing `sh`, so that sudo invokes the script as an executable (which will also honour any `inline_shebang` settings you have made)

Demonstration:

```
$ cat <<EOS >wibble.sh
#!/bin/sh -e
rm nonexistent
echo "shouldn't get here"
EOS
$ chmod +x wibble.sh
$ sudo sh ./wibble.sh
rm: nonexistent: No such file or directory
shouldn't get here
$ sudo ./wibble.sh
rm: nonexistent: No such file or directory
```

I realise the `-e` issue is described elsewhere in the page (under the `inline_shebang` setting), but I think it would be good for the sudo example to be correct. I was bitten by copy-pasting it as-is and assuming it was correct.

**Comment by mwhooker** [comment]
awesome catch, thanks!

**Comment by mwhooker** [comment]
looks like this was fixed in 4a4d529b

**Comment by candlerb** [comment]
Yes that works too, and avoids the need for `sudo -E`

There is another issue though. The documentation says:

> execute_command (string) - The command to use to execute the script. By default this is `chmod +x {{ .Path }}; {{ .Vars }} {{ .Path }}`.

The implication is that the `.Path` file cannot be relied upon to have the execute flag set. If that's true, shouldn't the sudo example also do this?

It wouldn't work if the execute flag wasn't set:

~~~
$ cat <<EOS >wibble.sh
#!/bin/sh -e
rm nonexistent
echo "shouldn't get here"
EOS
$ chmod -x wibble.sh
$ sudo -S sh -c './wibble.sh'
Password:
sh: ./wibble.sh: Permission denied
~~~
